dsm/resource_sobject.go

```diff
@@ -163,6 +169,15 @@ func contains(s []string, str string) bool {
 	return false
 }
 
+// this function takes a string in JSON format and unmarshals it. If the string is not in correct JSON format, it returns nil.
+func unmarshalStringToJson(inputString string) interface{}{
```

Comment:
Why? Why destroy the error message? Why not just handle this and return the error any time you need to handle unmarshaling json?

Reply:
Now it will return the actual error.